pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint_test.go

```diff
@@ -181,6 +185,44 @@ func TestGetRemoteEntrypoint(t *testing.T) {
 	}
 }
 
+func TestGetImageDigest(t *testing.T) {
+	img := getImage(t, &v1.ConfigFile{
+		ContainerConfig: v1.Config{},
+	})
+	digestHash, _ := img.Digest()
+	expectedDigetsSha := digestHash.String()
+	expectedRepo := "image"
+	manifestPath := fmt.Sprintf("/v2/%s/manifests/latest", expectedRepo)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
this is for my own understanding: how does `httptest` intercept requests?? i dont see anywhere where it is configured to listen on a particular port, or anywhere that we tell `remote.Image` to make requests to this server 🤔  (obviously it works but I don't understand how!)

Reply:
`httptest.NewServer` creates a thing (listening to a port or a socket I don't remember) and to target it, `server.URL` will point to it. (See https://golang.org/src/net/http/httptest/server.go?s=2302:2346#L70, it runs an `http.Server` behind the scene :angel: )

```go
func newLocalListener() net.Listener {
	if *serve != "" {
		l, err := net.Listen("tcp", *serve)
		if err != nil {
			panic(fmt.Sprintf("httptest: failed to listen on %v: %v", *serve, err))
		}
		return l
	}
	l, err := net.Listen("tcp", "127.0.0.1:0")
	if err != nil {
		if l, err = net.Listen("tcp6", "[::1]:0"); err != nil {
			panic(fmt.Sprintf("httptest: failed to listen on a port: %v", err))
		}
	}
	return l
}
```

Reply:
Oooooooooh and then we use `server.URL` in `image := path.Join(strings.TrimPrefix(server.URL, "http://"), "image:latest")` ! Got it, thanks for explaining @vdemeester :D